### PR TITLE
Formplayer middleware

### PR DESCRIPTION
**Pull request by benrudolph**
@wpride @biyeun this adds the ability to add middlewares to routes locally. this'll save us from doing tedious copy and pastes into every route. used it to hide the back button on the home page. also got rid of a ton of calls to `clearForm`. putting on label because right now form entry is broken and i want to test it with submissions

cc: @emord @nickpell

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js, line 238 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js, line 244 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/session_middleware.js, line 3 - 'Marionette' is defined but never used (no-unused-vars)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/session_middleware.js, line 13 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/session_middleware.js, line 16 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/session_middleware.js, line 26 - Unexpected console statement. (no-console)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/session_middleware.js, line 28 - 'name' is defined but never used (no-unused-vars)

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/session_middleware.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*global FormplayerFrontend */
+
+FormplayerFrontend.module("SessionNavigate", function (SessionNavigate, FormplayerFrontend, Backbone, Marionette) {
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/session_middleware.js]: 'Marionette' is defined but never used (no-unused-vars)

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/session_middleware.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    };
+    var backButtonShowHideMiddleware = function(name) {
+        if (name === 'singleApp') {
+            FormplayerFrontend.trigger('phone:back:hide');
+        } else {
+            FormplayerFrontend.trigger('phone:back:show');
+        }
+    };
+    var logRouteMiddleware = function(name) {
+        console.log('User navigated to ' + name);
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/session_middleware.js]: Unexpected console statement. (no-console)

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/session_middleware.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    };
+    var clearFormMiddleware = function(name) {
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/session_middleware.js]: 'name' is defined but never used (no-unused-vars)

**Comment by benrudolph** [comment]
tested it with formplayer patch and this is g2g, leaving the rest of dimagimon errors

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/session_middleware.js (hunk 2 above)
Comment (emord) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/session_middleware.js]: I think you can get around this by using `window.console.log` there's a couple of those in vellum

**Pull request merged**